backstory/src/main/java/com/google/sps/servlets/AnalyzeImageServlet.java

```diff
@@ -0,0 +1,311 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Text;
+import com.google.protobuf.ByteString;
+import com.google.sps.images.ImagesManager;
+import com.google.sps.images.VisionImagesManager;
+import com.google.sps.images.data.AnnotatedImage;
+import com.google.sps.perspective.PerspectiveStoryAnalysisManager;
+import com.google.sps.perspective.StoryAnalysisManager;
+import com.google.sps.perspective.data.APINotAvailableException;
+import com.google.sps.perspective.data.NoAppropriateStoryException;
+import com.google.sps.perspective.data.StoryDecision;
+import com.google.sps.story.PromptManager;
+import com.google.sps.story.StoryManager;
+import com.google.sps.story.StoryManagerImpl;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Backend servlet which manages the analysis of images, creation of backstories, and uploading
+ * the analyzed images with the backstory to permanent storage.
+ */
+@WebServlet("/analyze-image")
+public class AnalyzeImageServlet extends HttpServlet {
+  private ImagesManager imagesManager;
+  private Boolean useMockStoryManager;
+  private StoryManager storyManager;
+  private StoryAnalysisManager storyAnalysisManager;
+  private DatastoreService datastore;
+  private Boolean useMockDatastoreService;
+  private BlobstoreService blobstoreService;
+  private Boolean useMockBlobstoreService;
+  private Entity mockAnalyzedImageEntity;
+
+  public AnalyzeImageServlet() throws IOException, APINotAvailableException {
+    imagesManager = new VisionImagesManager();
+    useMockStoryManager = false;
+    storyAnalysisManager = new PerspectiveStoryAnalysisManager();
+    useMockDatastoreService = false;
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    useMockBlobstoreService = false;
+    blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
```

Comment:
I'm not sure it's safe to set these as fields... it would likely be safer to create new instances on each invocation of the POST request. The reason for this is that it seems possible that the same servlet instance may be accessed from different threads (see https://stackoverflow.com/a/12509971), and I'm not sure all of these are thread-safe.

It would probably be safer to have factories for these. If you don't want to create a factory interface+implementation, then you could probably use an interface+lambda.

```
interface StoryManagerFactory {
  StoryManager newInstance();
}

class AnalyzeImageServlet extends HTTPServlet {
  private StoryManagerFactory storyManagerFactory;

  public AnalyzeImageServlet() {
    storyManagerFactory = () -> { return new StoryManager(); }
  }

  public void setStoryManagerFactory(StoryManagerFactory storyManagerFactory) {
    this.storyManagerFactory = storyManagerFactory;
  }
}

class Test {
  void test() {
    AnalyzeImageServlet servlet = AnalyzeImageServlet();
    StoryManager mockStoryManager = mock(...);
    servlet.setStoryManagerFactory(() -> { return mockStoryManager; });
  }
}
```
See:
https://stackoverflow.com/a/4685606